drivers/flash/flash_stm32_fmc_mt29f4g08.c

```diff
@@ -0,0 +1,338 @@
+		struct nand_flash_address address =
+			flash_mt29f4g08_calculate_address(config, offset);
+		int ret = flash_stm32_fmc_nand_write_page(controller, &address,
+							  (const uint8_t *)data);
+		if (ret != 0) {
+			LOG_ERR("Writing page %d at block %d/plane %d failed with error %d",
+				address.page, address.block, address.plane, ret);
+			return ret;
+		}
+		data = (const uint8_t *)data + config->page_size;
+		offset += config->page_size;
+		len -= config->page_size;
+	}
+
+	return 0;
+}
+
+static int flash_mt29f4g08_erase(const struct device *dev, off_t offset, size_t size)
+{
+	const struct flash_mt29f4g08_config *config = dev->config;
+	const struct device *controller = config->controller;
+	int ret;
+
+	if ((offset < 0) || (offset >= config->flash_size) ||
+	    (size > (config->flash_size - offset))) {
+		return -EINVAL;
+	}
+
+	if (((offset % config->block_size) != 0) || ((size % config->block_size) != 0)) {
+		return -EINVAL;
+	}
+
+	while (size > 0) {
+		struct nand_flash_address address =
+			flash_mt29f4g08_calculate_address(config, offset);
+
+		ret = flash_stm32_fmc_nand_erase_block(controller, &address);
+		if (ret != 0) {
+			LOG_ERR("Erasing block %d at plane %d failed with error %d", address.block,
+				address.plane, ret);
+			return ret;
+		}
+		offset += config->block_size;
+		size -= config->block_size;
+	}
+
+	return 0;
+}
+
+static const struct flash_parameters *flash_mt29f4g08_get_parameters(const struct device *dev)
+{
+	const struct flash_mt29f4g08_config *config = dev->config;
+
+	return &config->parameters;
+}
+
+static int flash_mt29f4g08_get_size(const struct device *dev, uint64_t *size)
+{
+	const struct flash_mt29f4g08_config *config = dev->config;
+
+	*size = config->flash_size;
+
+	return 0;
+}
+
+#if CONFIG_FLASH_PAGE_LAYOUT
+static void flash_mt29f4g08_page_layout(const struct device *dev,
+					const struct flash_pages_layout **layout,
+					size_t *layout_size)
+{
+	const struct flash_mt29f4g08_config *config = dev->config;
+
+	*layout = &config->layout;
+	*layout_size = 1;
+}
+#endif /* CONFIG_FLASH_PAGE_LAYOUT */
+
+#if CONFIG_FLASH_EX_OP_ENABLED
+int flash_mt29f4g08_ex_op(const struct device *dev, uint16_t code, const uintptr_t in, void *out)
+{
+	const struct flash_mt29f4g08_config *config = dev->config;
+	int ret = 0;
+
+	switch (code) {
+	case NAND_FLASH_IS_BAD_BLOCK: {
+		/* Check bad block in first page */
+		size_t block = *(size_t *)in;
+		struct nand_flash_address address;
+		uint8_t spare_area[config->spare_area_size];
+
+		if (block * config->block_size >= config->flash_size) {
+			*(int *)out = NAND_BLOCK_BAD;
+			return -EINVAL;
+		}
+
+		memset(spare_area, 0x00, config->spare_area_size);
+		address = flash_mt29f4g08_calculate_address(config, block * config->block_size);
+		ret = flash_stm32_fmc_nand_read_spare_area(config->controller, &address,
+							   spare_area);
+		*(int *)out =
+			((ret == 0) && (spare_area[0] != 0xFF)) ? NAND_BLOCK_BAD : NAND_BLOCK_GOOD;
+		break;
+	}
```

Comment:
I don't think these braces (in both `case` sequences)  are useful. There indentation could be confusing. Could you remove them?